sos/report/plugins/ceph_mgr.py

```diff
@@ -6,28 +6,68 @@
 #
 # See the LICENSE file in the source distribution for further information.
 
+import os
+
 from sos.report.plugins import Plugin, RedHatPlugin, UbuntuPlugin
 
 
 class CephMGR(Plugin, RedHatPlugin, UbuntuPlugin):
+    """
+    This plugin is for capturing information from Ceph mgr nodes. While the
+    majority of this plugin should be version-agnostic, several collections are
+    dependent upon the version of Ceph installed. Versions that correlate to
+    RHCS 4 or RHCS 5 are explicitly handled for differences such as those
+    pertaining to log locations on the host filesystem.
+
+    Note that while this plugin will activate based on the presence of Ceph
+    containers, commands are run directly on the host as those containers are
+    often not configured to successfully run the `ceph` commands collected by
+    this plugin. These commands are majorily `ceph daemon` commands that will
+    reference discovered admin sockets under /var/run/ceph.
+    """
 
     short_desc = 'CEPH mgr'
 
     plugin_name = 'ceph_mgr'
     profiles = ('storage', 'virt', 'container')
-    files = ('/var/lib/ceph/mgr/',)
-    containers = ('ceph-(.*)?mgr.*',)
+    files = ('/var/lib/ceph/mgr/', '/var/lib/ceph/*/mgr*')
+    containers = ('ceph-(.*-)?mgr.*',)
 
     def setup(self):
+
+        self.ceph_version = self.get_ceph_version()
+
+        logdir = '/var/log/ceph'
+        libdir = '/var/lib/ceph'
+        rundir = '/run/ceph'
+
+        if self.ceph_version >= 16:
+            logdir += '/*'
+            libdir += '/*'
+            rundir += '/*'
+
         self.add_file_tags({
-            '/var/log/ceph/ceph-mgr.*.log': 'ceph_mgr_log',
+            f'{logdir}/ceph-mgr.*.log': 'ceph_mgr_log',
         })
 
+        self.add_forbidden_path([
+            "/etc/ceph/*keyring*",
+            f"{libdir}/*keyring*",
+            f"{libdir}/**/*keyring*",
+            f"{libdir}/osd*",
+            f"{libdir}/mon*",
+            # Excludes temporary ceph-osd mount location like
+            # /var/lib/ceph/tmp/mnt.XXXX from sos collection.
+            f"{libdir}/tmp/*mnt*",
+            "/etc/ceph/*bindpass*",
+        ])
+
         self.add_copy_spec([
-            "/var/log/ceph/ceph-mgr*.log",
-            "/var/lib/ceph/mgr/",
-            "/var/lib/ceph/bootstrap-mgr/",
-            "/run/ceph/ceph-mgr*",
+            f"{logdir}/ceph-mgr*.log",
+            f"{libdir}/mgr/",
+            f"{libdir}/mgr*/",
```

Comment:
The `/mgr/` is dedicatedly mentioned due to sizelimit? Since that `copy_spec` would be already contained in `mgr*/` otherwise.

Reply:
Nope, that's an oversight.